src/dlgPackageExporter.cpp

```diff
@@ -136,15 +150,49 @@ void dlgPackageExporter::recurseTree(QTreeWidgetItem* pItem, QList<QTreeWidgetIt
     }
 }
 
+bool dlgPackageExporter::writeFileToZip(const QString& archiveFileName, const QString& fileSystemFileName, zip* archive)
+{
+    struct zip_source* s = zip_source_file(archive, fileSystemFileName.toUtf8().constData(), 0, -1);
+    if (s == nullptr) {
+        displayResultMessage(tr("Failed to source file \"%1\" to place into (.zip format archive) package file, error is: \"%3\".",
+                                // Intentional comment to separate arguments
+                                "This error message will appear when a file is to be placed into the .zip type file (though we give it an .mpackage extension) but the libzip library cannot open it.")
```

Comment:
I will send a PR to your branch to make this easier.

Reply:
The niggle with that it that it is not clear that it is actually *opening* the file at that point - it *seems* that the libzip code is making some sort of record of the file to be added (*sourced*?) - *but it does not actually get processed until the zip file is __zip_close()__ ed at the end of the method* - which (as I have pointed out elsewhere) can take a noticeable time.

Reply:
You see, for me, as a non-native, I wouldn't even know what *niggle* means, let alone how to translate it, or to come back closer to the matter at hand, a "source file" is a very well-defined concept, but "Failed to source file" leaves me puzzled. 

What I am trying to say is: There is a difference between the level of detail necessary for a mere user using a software, and the developer developing it. The delicacies of libzip library would fall into the latter category for me. Something is wrong with the %1 file which was intended to be opened at some point in this operation. That seems about enough info. 

Also I changed %3 to %2 in this message. Again, all details in said PR linked to below.

Reply:
:book: Actually Qt will handle the absence of a %2 just fine - it automagically looks to replace the lowest numbered *unhandled* parameter one by one and will only throw a fit if there is not enough arguments by the time it gets to the last one (can't recall how it behaves if there are *too many* arguments - it might get unhappy as well then or it might just ignore the excess...) - OTOH I will tweak that before I merge this as it otherwise is...